app/src/main/java/com/nononsenseapps/feeder/ui/compose/searchfeed/SearchFeedViewModel.kt

```diff
@@ -35,55 +37,96 @@ class SearchFeedViewModel(di: DI) : DIAwareViewModel(di) {
     )
 
     fun searchForFeeds(initialUrl: URL): Flow<Either<FeedParserError, SearchResult>> {
-        return flow {
-            siteMetaData = feedParser.getSiteMetaData(initialUrl)
-            // Flow collection makes this block concurrent with map below
-            val initialSiteMetaData = siteMetaData
-
-            initialSiteMetaData
-                .onRight { metaData ->
-                    metaData.alternateFeedLinks.forEach {
-                        emit(Either.Right(it.link))
+        return if (initialUrl.isAdaptedUrlFromNostrUri()) {
+            flow {
+                val nostrProfile = feedParser.getProfileMetadata(initialUrl)
+                nostrProfile
+                    .onRight {
+                        emit(Either.Right(it))
                     }
-                    if (metaData.alternateFeedLinks.isEmpty()) {
-                        emit(Either.Left(NoAlternateFeeds(initialUrl.toString())))
+                    .onLeft {
+                        emit(Either.Left(it))
                     }
-                }
-                .onLeft {
-                    if (it is HttpError) {
-                        handleHttpError(it)
+            }.map { profileResult ->
+                profileResult.flatMap { metadata ->
+
+                    Either.catching(
+                        onCatch = { t -> FetchError(url = metadata.uri, throwable = t) },
+                    ) {
+                        SearchResult(
+                            title = metadata.name,
+                            url = metadata.uri,
+                            description = "Nostr Articles by ${metadata.name}",
```

Comment:
Is this user visible? If so should not be hard coded english

Reply:
Fixed.